Tony/8 about button

tonypioneer opened the pull request:
# Pull Request Details

## What issue does this PR address

- Link to associated issue: #8 #9

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [ ] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [ ] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev

gjwgit commented on lib/src/widgets/solid_about_button.dart:
```diff
@@ -0,0 +1,245 @@
+        applicationName: applicationName,
+        applicationVersion: applicationVersion,
+        applicationIcon: config.applicationIcon,
+        applicationLegalese: config.applicationLegalese ??
```
Perhaps we could use wordWrap() on the legalese by default.

tonypioneer replied:
Wrap() has been added to the legalese.

gjwgit reviewed (changes requested):
Sorry for the confusion.

I was not asking to use the Wrap widget.

Instead I simply wanted the legalese string to be word wrapped using wordWrap() from markdown_tooltip.

The current version using Wrap results in this:

<img width="1332" height="819" alt="Screenshot From 2025-08-23 08-18-39" src="https://github.com/user-attachments/assets/54c0267d-6748-4d86-b954-3cd1ef3febe0" />

Without using wordWrap() it retains the word wrapping and space indentation of the original text:
<img width="1332" height="819" alt="Screenshot From 2025-08-23 08-15-11" src="https://github.com/user-attachments/assets/9b1c5074-943c-4869-97ad-f87c1ae625b7" />

What I wanted was for solidui to do wordWrap() that wraps the text to a certain width (the default is okay) and removes the indetation space. In this example I use wordWrap() to do that in the app. But would be good to have the wrodWrap in solidui rather than having to do it in the app:

<img width="1332" height="819" alt="Screenshot From 2025-08-23 08-13-21" src="https://github.com/user-attachments/assets/9654677e-abfc-4265-a00d-76916604ecb6" />

gjwgit commented:
I think line 214 of solid_about_button,dart could become:

```
    final legalText = wordWrap(config.applicationLegalese ??
        '© ${DateTime.now().year} $applicationName';)
```

And then use your previous code without the Wrap widget.

tonypioneer commented:
Thanks @gjwgit for the clarification. I'll adjust the code.

gjwgit reviewed (commented):
Thanks @tonypioneer.

Another request, instead of `children` can we have `text` so that I simply provide the text for the ABOUT rather than a collection of widgets.

So, from my developer point of view, I do:

```
      aboutConfig: SolidAboutConfig(
        applicationName: 'MovieStar',
        applicationIcon: const Icon(
          Icons.movie,
          size: 64,
        ),
        applicationLegalese: '''
        © ${DateTime.now().year} Software Innovation Institute, ANU
        ''',
        text: '''

MovieStar is a movie discovery and management application built with
Flutter, which helps you discover, track, and manage your favourite
movies using the power of Solid PODs for decentralised data storage.

Licensed under the GNU General Public License v3.0

This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Visit https://github.com/anusii/moviestar for more information.

''',

Then similarly do wordWrap() on the text and place this inside a MarkdownBody()

```
                  children: [
                    MarkdownBody(
                      data: wordWrap(config.text),
                      selectable: true,
                      softLineBreak: true,
                      onTapLink: (text, href, about) {
                        final Uri url = Uri.parse(href ?? '');
                        launchUrl(url);
                      },
                    ),
                  ],
```
      ),
```

tonypioneer commented:
> Thanks @tonypioneer.
> 
> Another request, instead of `children` can we have `text` so that I simply provide the text for the ABOUT rather than a collection of widgets.
> 
> So, from my developer point of view, I do:
> 
> ```
>       aboutConfig: SolidAboutConfig(
>         applicationName: 'MovieStar',
>         applicationIcon: const Icon(
>           Icons.movie,
>           size: 64,
>         ),
>         applicationLegalese: '''
>         © ${DateTime.now().year} Software Innovation Institute, ANU
>         ''',
>         text: '''
> 
> MovieStar is a movie discovery and management application built with
> Flutter, which helps you discover, track, and manage your favourite
> movies using the power of Solid PODs for decentralised data storage.
> 
> Licensed under the GNU General Public License v3.0
> 
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> 
> Visit [github](https://github.com/anusii/moviestar) for more information.
> 
> ''',
> 
> Then similarly do wordWrap() on the text and place this inside a MarkdownBody()
> ```
> 
> ```
>               children: [
>                 MarkdownBody(
>                   data: wordWrap(config.text),
>                   selectable: true,
>                   softLineBreak: true,
>                   onTapLink: (text, href, about) {
>                     final Uri url = Uri.parse(href ?? '');
>                     launchUrl(url);
>                   },
>                 ),
>               ],
> ```
> 
> ```
>       ),
> ```

Hi @gjwgit, this feature has been implemented.

<img width="1118" height="758" alt="image" src="https://github.com/user-attachments/assets/4351f6e6-dfa2-4426-804d-60ca27b28884" />

gjwgit commented:
Thanks Tony - The `text` should not be indented/padded.

<img width="1332" height="819" alt="image" src="https://github.com/user-attachments/assets/7c3465eb-5c44-4672-b78c-b52ec0f71e9c" />

I modified - please check it looks okay.

tonypioneer commented:
> Thanks Tony - The `text` should not be indented/padded.
> 
> <img alt="image" width="1332" height="819" src="https://private-user-images.githubusercontent.com/1681429/481415895-7c3465eb-5c44-4672-b78c-b52ec0f71e9c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYwNzU0MjQsIm5iZiI6MTc1NjA3NTEyNCwicGF0aCI6Ii8xNjgxNDI5LzQ4MTQxNTg5NS03YzM0NjVlYi01YzQ0LTQ2NzItYjc4Yy1iNTJlYzBmNzFlOWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjRUMjIzODQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmJhZTE3NDYyZjE2ZjMwYWI0YmNjYTRhNTU2NGJkYjcyMTM3NjM2NzM2YTUyOTJiNzYyMGJkODc5MTM2YjU1MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.f-cw6FaBRfdpkSlo4EGbT5C6U-VUhiKRVHdEfTciyLc">
> I modified - please check it looks okay.

Thanks @gjwgit. Yes, in the previous commit, I added some indentation to the left of the text like this.

<img width="1118" height="758" alt="image" src="https://github.com/user-attachments/assets/4546cd08-b110-4cb4-b348-7214557b3465" />

But it looks good if the indentation is removed.

gjwgit commented:
Perfect. Thanks @tonypioneer.
